apiserver/facades/agent/storageprovisioner/storageprovisioner.go

```diff
@@ -882,7 +891,7 @@ func (s *StorageProvisionerAPIv4) FilesystemParams(ctx context.Context, args par
 	if err != nil {
 		return params.FilesystemParamsResults{}, err
 	}
-	controllerCfg, err := s.st.ControllerConfig()
+	controllerCfg, err := s.controllerConfigGetter.ControllerConfig(ctx)
```

Comment:
Since you only ever need this to get the controller UUID, and that will never change, you might as well get it once on first use and cache it.